src/Microsoft.DotNet.Interactive/Kernel.cs

```diff
@@ -752,6 +751,64 @@ public void RegisterForDisposal(IDisposable disposable)
         }
     }
 
+    private Task PublishDirectiveHoverTextAsync(
+        RequestHoverText command,
+        KernelInvocationContext context)
+    {
+        if (command.SyntaxNode is DirectiveNode directiveNode)
+        {
+            string hoverText = null;
+
+            if (!directiveNode.TryGetDirective(out var directive))
+            {
+                return Task.CompletedTask;
+            }
+
+            var node = directiveNode.FindNode(command.OriginalPosition);
+
+            switch (node)
+            {
+                case DirectiveNameNode { Parent: DirectiveSubcommandNode subcommandNode }:
+                    if (subcommandNode.TryGetSubcommand(out var subcommandDirective))
+                    {
+                        hoverText = subcommandDirective.Description;
+                    }
+
+                    break;
+
+                case DirectiveNameNode _:
+                    hoverText = directive.Description;
+                    break;
+
+                case DirectiveParameterNameNode directiveParameterNameNode:
+                    if (directiveParameterNameNode.Parent is DirectiveParameterNode pn &&
+                        pn.TryGetParameter(out var parameter))
+                    {
+                        hoverText = parameter.Description;
+                    }
+
+                    break;

```

Comment:
Should we add a `defualt:` case to throw for unexpected node type?

Reply:
No. It's not an exceptional case. Some node types don't (currently) support hover text.